server.py

```
@@ -59,35 +61,96 @@ def main():
+                        with open('/tmp/bbs_heartbeat', 'w') as f:
+                            f.write(str(time.time()))
+                    except Exception:
+                        pass
+
+                    # 2. Aggressive socket watchdog for TCP interfaces
+                    if system_config['interface_type'] == 'tcp' and hasattr(interface, 'socket') and interface.socket:
+                        try:
+                            # Using b"" is safe for Meshtastic framing but triggers OS-level BrokenPipe check
+                            interface.socket.send(b"", socket.MSG_DONTWAIT)
+                        except (BrokenPipeError, OSError):
+                            logging.error("Detected BrokenPipe in underlying TCP socket watchdog.")
```

Ruff (0.15.1)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)